pkg/k8sapi/k8sutils.go

```diff
@@ -120,27 +122,30 @@ func CreateKubeClient(appName string) (client.Client, error) {
 	}
 	cacheReader, err := CreateKubeClientCache(restCfg, vpcCniScheme, filterMap)
 	if err != nil {
-		return nil, err
+		log.Warnf("Skipping cache-based Kubernetes client: %s", err)
+		cacheReader = nil
```

Comment:
Just curiosity, if error is returned, the reader is not nil?

Reply:
if apiserver is not available, CreateKubeClientCache() will fail because it cannot fetch info via NewDynamicRESTMapper(), so error is returned. and cacheReader is nil in this case.
basically no apiserver = no cache.